texture-font.c

```diff
@@ -381,28 +393,54 @@ texture_font_load_face( texture_font_t *self, float size )
 	    }
 	    break;
 	}
-    }
 
-    /* Select charmap */
-    error = FT_Select_Charmap(self->face, FT_ENCODING_UNICODE);
-    if(error) {
-        freetype_error( error, "FT_Error (%s:%d, code 0x%02x) : %s\n",
-			__FILENAME__, __LINE__, FT_Errors[error].code, FT_Errors[error].message);
-        goto cleanup_face;
-    }
+	/* Select charmap */
+	error = FT_Select_Charmap(self->face, FT_ENCODING_UNICODE);
+	if(error) {
+	    freetype_error( error, "FT_Error (%s:%d, code 0x%02x) : %s\n",
+			    __FILENAME__, __LINE__, FT_Errors[error].code, FT_Errors[error].message);
+	    goto cleanup_face;
+	}
 
-    /* Set char size */
-    error = FT_Set_Char_Size(self->face, (int)(size * HRES), 0, DPI * HRES, DPI);
-    
-    if(error) {
-        freetype_error( error, "FT_Error (%s:%d, code 0x%02x) : %s\n",
-			__FILENAME__, __LINE__, FT_Errors[error].code, FT_Errors[error].message);
-        goto cleanup_face;
+	if( texture_is_color_font( self ) ) {
+	    /* Select best size */
+	    if (self->face->num_fixed_sizes == 0) {
+		freetype_error( error, "FT_Error (%s:%d) : no fixed size in color font\n",
+				__FILENAME__, __LINE__);
+		goto cleanup_face;
+	    }
+	
+	    int best_match = 0;
+	    int diff = abs((int)size - self->face->available_sizes[0].width);
+
+	    for (int i = 1; i < self->face->num_fixed_sizes; ++i) {
+		int ndiff = abs((int)size - self->face->available_sizes[i].width);
```

Comment:
This is resulting in emojis that are a bit too big: 
![image](https://user-images.githubusercontent.com/184008/119440483-bf146c80-bd67-11eb-9b8c-b1d48fce80c1.png)

May I suggest only considering sizes that are less than or equal to the requested size? Also, is that i = 1 initial assignment deliberate? Seems odd.

```suggestion
	    for (i = 0; i < self->face->num_fixed_sizes; ++i) {
		if ( self->face->available_sizes[i].width > size ) continue;
		int ndiff = abs((int)size - self->face->available_sizes[i].width);
```

![image](https://user-images.githubusercontent.com/184008/119440626-026edb00-bd68-11eb-94a3-3a88d2279a42.png)

Reply:
Yes, the `i=1` is intentional. I start with `best_match=0` (that's the index of the best match) and then try to find a better match.  There's no point to look at `i=0` again.  There's also no guarantee that there is a size available smaller than the one you requested.

The size chosen is the one with the smallest difference to the requested size, and you need to use the value of the new `self->scale` field to draw it correctly scaled.  It's set to 1.0 for normal fonts, so you can use this scaling factor all the time.

Reply:
Ah! That makes sense!

Reply:
The selection algorithm is from here.

https://github.com/hironishihara/ofxMixedFont/blob/master/src/ofxFT2Font.cpp